lib/plugins/aws/package/compile/events/alb/lib/resolveLambdaTarget.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+module.exports = {
```

Comment:
It would nice to put it here: https://github.com/serverless/serverless/tree/ca693872855a59799ec22079d20d048b40ab33a1/lib/plugins/aws/utils, export the function directly, memoize it as here: https://github.com/serverless/serverless/blob/9b701a405627273fb54e411eb4e87bc085282c6b/lib/plugins/aws/package/compile/events/httpApi/index.js#L11-L15 and rely on it, in all three places (APIGW permissions, here and HTTP API) where this target is resolved now.

We may also let function accept function object directly (not function name), I think it'll make it easier for internals of a functions and its consumers

Reply:
Sorry for the late reply, havn't had much time to work on this lately.
The value of `functionLogicalId` isn't in the `functionObject` properties, so I would have to pass it as a second parameter, like this : 
```javascript
 const resolveTargetConfig = memoizee((functionObject, functionLogicalId) => { 
   const functionArnGetter = { 'Fn::GetAtt': [functionLogicalId, 'Arn'] }; 
   if (!functionObject.targetAlias) return functionArnGetter; 
   return { 'Fn::Join': [':', [functionArnGetter, functionObject.targetAlias.name]] }; 
 }); 
```
Is is OK for you @medikoo ?

Reply:
> The value of functionLogicalId isn't in the functionObject properties, so I would have to pass it as a second parameter, like this :

Fortunately I think we don't have to do that, as we can resolve it from `functionObject` as following:

```javascript
const naming = require('../lib/naming');
...
const functionLogicalId = naming.getLambdaLogicalId(functionObject.name);
```

Reply:
Looks like `functionObject.name` equals to `${that.service}-${stageNameForFunction}-${functionName}` so it can't be used with `getLambdaLogicalId()` unless I make some hacky code to extract the function name.. 

So I went for :
```javascript
const resolveLambdaTarget = memoizee((functionObject, functionName) => {
  const lambdaLogicalId = naming.getLambdaLogicalId(functionName);
  const functionArnGetter = { 'Fn::GetAtt': [lambdaLogicalId, 'Arn'] };
  if (!functionObject.targetAlias) return functionArnGetter;
  return { 'Fn::Join': [':', [functionArnGetter, functionObject.targetAlias.name]] };
});
```

Reply:
Indeed, I forgot it's already resolved one for CF template needs.

What you propose looks great. I would just probably switch order of arguments (it feels more natural and composable to have `key, value` instead of `value, key`)

Reply:
it's done 👍